interest-rate-models/test/unit/fixed_rate.test.ts

```diff
@@ -0,0 +1,139 @@
+            inputAssets: [{ address: testAddress, asset: { alphAmount: 10n ** 18n } }]
+        }
+    })
+
+    it('test getRate', async () => {
+        const testParams = {
+            ...testParamsFixture,
+            testArgs: {}
+        }
+
+        const testResult = await FixedRate.tests.getRate(testParams)
+
+        // Check the return value
+        expect(testResult.returns).toEqual(100000000000000000n)
+
+        // Verify contract state remains unchanged
+        const contractState = testResult.contracts[0] as FixedRateTypes.State
+        expect(contractState.fields.rate).toEqual(100000000000000000n)
+        expect(contractState.fields.rateUpdated).toEqual(false)
+        expect(contractState.fields.admin).toEqual(testAddress)
+    })
+
+    it('test setBorrowRate success', async () => {
+        const testParams = {
+            ...testParamsFixture,
+            testArgs: { newBorrowRate: 50000000000000000n }
+        }
+
+        const testResult = await FixedRate.tests.setBorrowRate(testParams)
+
+        // Check the return value
+        expect(testResult.returns).toEqual(null)
+
+        // Verify contract state was updated
+        const contractState = testResult.contracts[0] as FixedRateTypes.State
+        expect(contractState.fields.rate).toEqual(50000000000000000n)
+        expect(contractState.fields.rateUpdated).toEqual(true)
+        expect(contractState.fields.admin).toEqual(testAddress)
+
+        // Verify a RateSet event was emitted
+        expect(testResult.events.length).toEqual(1)
+        const event = testResult.events[0] as FixedRateTypes.RateSetEvent
+        expect(event.name).toEqual('RateSet')
+        expect(event.fields.setter).toEqual(testAddress)
+        expect(event.fields.oldRate).toEqual(100000000000000000n)
+        expect(event.fields.newRate).toEqual(50000000000000000n)
+    })
+
+    it('test setBorrowRate fails when already updated', async () => {
+        const testParams = {
+            ...testParamsFixture,
+            initialFields: {
+                ...testParamsFixture.initialFields,
+                rateUpdated: true
+            },
+            testArgs: { newBorrowRate: 50000000000000000n }
+        }
+
+        // Test that assertion fails with the RateAlreadySet error code (0)
+        await expectAssertionError(
+            FixedRate.tests.setBorrowRate(testParams),
+            testContractAddress,
+            0 // RateAlreadySet error code
+        )
+    })
+
+    it('test setBorrowRate fails when rate is too high', async () => {
+        const testParams = {
+            ...testParamsFixture,
+            testArgs: { newBorrowRate: 2000000000000000000n } // 2.0 * 10^18 = 200% (too high)
+        }
+
+        // Test that assertion fails with the InvalidRate error code (1)
+        await expectAssertionError(
+            FixedRate.tests.setBorrowRate(testParams),
+            testContractAddress,
+            1 // InvalidRate error code
```

Comment:
Instead of 1 you can write the error code like this: `Number(FixedRate.consts.ErrorCodes.InvalidRate)`